src/mongo_logger.ts

```diff
@@ -122,58 +176,252 @@ function resolveLogPath(
   {
     mongodbLogPath
   }: {
-    mongodbLogPath?: unknown;
+    mongodbLogPath?: string | MongoDBLogWritable;
   }
-): Writable {
+): MongoDBLogWritable {
   const isValidLogDestinationString = (destination: string) =>
     ['stdout', 'stderr'].includes(destination.toLowerCase());
   if (typeof mongodbLogPath === 'string' && isValidLogDestinationString(mongodbLogPath)) {
-    return mongodbLogPath.toLowerCase() === 'stderr' ? process.stderr : process.stdout;
+    return (mongodbLogPath.toLowerCase() === 'stderr'
+      ? process.stderr
+      : process.stdout) as unknown as MongoDBLogWritable;
```

Comment:
This typecasting feels inelegant so I'm open to any suggestions on how to improve this.